modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/DiagramHeader.js

```diff
@@ -15,123 +15,143 @@ import ClayDropDown from '@clayui/drop-down';
 import ClayForm from '@clayui/form';
 import ClaySlider from '@clayui/slider';
 import PropTypes from 'prop-types';
-import React, {useState} from 'react';
+import React, {useContext, useState} from 'react';
 
-const DiagramHeader = ({
-	addNewPinState,
-	namespace,
-	newPinSettings,
-	radiusChoice,
-	setAddNewPinState,
-	setAddPinHandler,
-}) => {
+import DiagramCommonContext from './context/DiagramCommonContext';
+
+const RADIUS_CHOICE = [
+	{
+		label: 'Small',
+		value: 10,
+	},
+	{
+		label: 'Medium',
+		value: 20,
+	},
+	{
+		label: 'Large',
+		value: 30,
+	},
+];
+
+const DiagramHeader = ({newPinSettings}) => {
 	const [active, setActive] = useState(false);
 	const [customColors, setCustomColors] = useState(
 		newPinSettings.colorPicker.defaultColors
 	);
+	const {
+		addNewPinState,
+		isAdmin,
+		namespace,
+		setAddNewPinState,
+		setAddPinHandler,
+	} = useContext(DiagramCommonContext);
 
 	return (
 		<div className="d-flex diagram diagram-header justify-content-between">
-			<div className="d-flex text-align-center">
-				<label className="align-middle my-auto">
-					{Liferay.Language.get('diagram-pin')}
-				</label>
-
-				<ClayDropDown
-					active={active}
-					className="my-auto"
-					onActiveChange={setActive}
-					trigger={
-						<ClayButton
-							alt={Liferay.Language.get(
-								'click-to-select-custom-diameter'
-							)}
-							className="ml-3 select-diameter"
-							displayType="secondary"
-						>
-							{Liferay.Language.get('default-diameter')}
-						</ClayButton>
-					}
-				>
-					<ClayDropDown.ItemList className="diagram-header-picker">
-						<ClayDropDown.Group
-							header={Liferay.Language.get('select-color')}
-						>
-							<ClayColorPicker
-								colors={customColors}
-								label={Liferay.Language.get('custom-colors')}
-								max={100}
-								name={`${namespace}diagram-color-picker`}
-								onColorsChange={setCustomColors}
-								onValueChange={(item) => {
-									setAddNewPinState({
-										fill: item.replace('#', ''),
-										radius: addNewPinState.radius,
-									});
-								}}
-								showHex={true}
-								step={1}
-								title={Liferay.Language.get('custom-colors')}
-								useNative={newPinSettings.colorPicker.useNative}
-								value={addNewPinState.fill}
-							/>
-						</ClayDropDown.Group>
-					</ClayDropDown.ItemList>
-
-					<ClayDropDown.Divider />
+			{isAdmin && (
+				<div className="d-flex text-align-center">
+					<label className="align-middle my-auto">
+						{Liferay.Language.get('diagram-pin')}
+					</label>
 
-					<ClayDropDown.ItemList>
-						<ClayDropDown.Group
-							header={Liferay.Language.get('select-radius')}
-						>
-							{radiusChoice.map((item, i) => (
-								<ClayDropDown.Item
-									key={i}
-									onClick={() => {
+					<ClayDropDown
+						active={active}
+						className="my-auto"
+						onActiveChange={setActive}
+						trigger={
+							<ClayButton
+								alt={Liferay.Language.get(
```

Comment:
I think it can be an `aria-label` isn't? I never saw using `alt` attrs for buttons. See https://www.w3.org/TR/wai-aria-practices-1.1/examples/button/button.html